crate_universe/src/context/crate_context.rs

```diff
@@ -142,6 +142,9 @@ pub struct CommonAttributes {
     #[serde(skip_serializing_if = "SelectList::is_empty")]
     pub proc_macro_deps_dev: SelectList<CrateDependency>,
 
+    #[serde(skip_serializing_if = "SelectList::is_empty")]
+    pub all_optional_deps: SelectList<CrateDependency>,
```

Comment:
Instead of tracking `all_optional_deps`, can you either update `Dependency` to have an `optional` field that we can use when gathering workspace member deps, or can you track each kind of optional dependency separately?